alf/networks/value_networks.py

```diff
@@ -225,7 +222,9 @@ def __init__(self,
                                     a=-0.03, b=0.03)
 
         self._encoding_net = LSTMEncodingNetwork(
-            input_tensor_spec=self._processed_input_tensor_spec,
+            input_tensor_spec=input_tensor_spec,
```

Comment:
> The base class ValueNetwork also creates self._encoding_net. That does not seem right.

It will be discarded. This is also the case in `ActorDistributionNetwork`. This way we can save some code of ``forward`` and ``make_parallel`.

Reply:
Now extracted a base class so that the encoding network won't be made twice.